src/EventService.Validation/EventCategory/CreateEventCategoryRequestValidator.cs

```diff
@@ -0,0 +1,29 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.EventService.Data.Interfaces;
+using LT.DigitalOffice.EventService.Models.Dto.Requests;
+using LT.DigitalOffice.EventService.Validation.EventCategory.Interfaces;
+
+namespace LT.DigitalOffice.EventService.Validation.EventCategory;
+
+public class CreateEventCategoryRequestValidator : AbstractValidator<CreateEventCategoryRequest>, ICreateEventCategoryRequestValidator
+{
+  public CreateEventCategoryRequestValidator(
+    IEventRepository eventRepository,
+    ICategoryRepository categoryRepository,
+    IEventCategoryRepository eventCategoryRepository)
+  {
+    RuleFor(x => x.EventId)
+      .MustAsync(async (eventId, _) => await eventRepository.DoesExistAsync(eventId))
+      .WithMessage("This event doesn't exist")
+      .MustAsync(async (eventId, _) => await eventCategoryRepository.CountAsync(eventId) < 5)
+      .WithMessage("This event has 5 categories");
```

Comment:
mb "This event already has 5 categories"

Reply:
and stop at the end (for all)
"This event already has 5 categories."